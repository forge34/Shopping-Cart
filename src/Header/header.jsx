import style from "./header.module.css";
import PropTypes from "prop-types";

function Header({ ItemNumber }) {
  return (
    <div className={style.container}>
      <h1 className={style.heading}>Shopping Cart</h1>

      <button className={style.hbtn + " push"}>Home</button>
      <img className={style.icon} src="shopping_cart.svg"></img>
      <p>{ItemNumber ? ItemNumber : 0} </p>
    </div>
  );
}

Header.propTypes = {
  ItemNumber: PropTypes.number,
};

export default Header;
