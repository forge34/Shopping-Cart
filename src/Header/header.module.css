@import url("../common-styles/main.css");

.container {
  background-color: var(--dark-gray);
  display: flex;
  font-family: "Roboto", sans-serif;
  align-items: center;
  justify-content: center;
  color: white;
  padding: 0 1.5em;
}

.icon {
    filter: invert(100%);
    margin-left: 1.25em;
}

.hbtn {
  background-color: var(--light-green);
  color: white;
  composes: btn;
  padding: 0.3em 1.2em;
  border-radius: 4px;
  font-size: 1.25em;
  height: 100%;
  margin-right: 3em;
}

.hbtn:hover {
  opacity: 70%;
}

.heading {
  margin: 0.3em 0.1em;
}
