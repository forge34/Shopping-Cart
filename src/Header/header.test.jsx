import { render, screen } from "@testing-library/react";
import { describe, expect } from "vitest";
import Header from "./header";

describe("Header component", () => {
  it("Renders Header navbar", () => {
    const header = render(<Header></Header>);

    expect(header).toMatchSnapshot();
  });

  it("Renders  item number", () => {
    render(<Header ItemNumber={30}></Header>);

    expect(screen.getByText("30")).toBeInTheDocument();
  });
});
