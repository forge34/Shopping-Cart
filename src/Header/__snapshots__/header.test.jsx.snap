// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > Renders Header navbar 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_container_aebbd9"
      >
        <h1
          class="_heading_aebbd9"
        >
          Shopping Cart
        </h1>
        <button
          class="_hbtn_aebbd9 push"
        >
          Home
        </button>
        <img
          class="_icon_aebbd9"
          src="shopping_cart.svg"
        />
        <p>
          0
           
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_container_aebbd9"
    >
      <h1
        class="_heading_aebbd9"
      >
        Shopping Cart
      </h1>
      <button
        class="_hbtn_aebbd9 push"
      >
        Home
      </button>
      <img
        class="_icon_aebbd9"
        src="shopping_cart.svg"
      />
      <p>
        0
         
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
